bindings.c

```diff
@@ -4092,6 +4292,363 @@ static int proc_swaps_read(char *buf, size_t size, off_t offset,
 	return rv;
 }
 
+/*
+ * Find the process pid from cgroup path.
+ * eg:from /sys/fs/cgroup/cpu/docker/containerid/cgroup.procs to find the process pid.
+ * @pid_buf : put pid to pid_buf.
+ * @dpath : the path of cgroup. eg: /docker/containerid or /docker/containerid/child-cgroup ...
+ * @depth : the depth of cgroup in container.
+ * @sum : return the number of pid.
+ * @cfd : the file descriptor of the mounted cgroup. eg: /sys/fs/cgroup/cpu
+ */
+static int calc_pid(char ***pid_buf, char *dpath, int depth, int sum, int cfd)
+{
+	DIR *dir;
+	int fd;
+	struct dirent *file;
+	FILE *f = NULL;
+	size_t linelen = 0;
+	char *line = NULL;
+	int pd;
+	char *path_dir, *path;
+	char **pid;
+
+	/* path = dpath + "/cgroup.procs" + /0 */
+	path = (char *)malloc(strlen(dpath) + 20);
```

Comment:
Please don't cast the result of `malloc()`.

Reply:
Sorry, my English is a bit poor. I didn’t understand the advice you gave me before about __cast__.
Now I understand.  Don't cast the result of __malloc()__ is not the same thing I learned in the course and is very cool. I will rewrite them. 
Thank you! @brauner

Reply:
Whether to cast the result of `malloc()` or not is a worthwhile discussion to have. :) We have never cast the result of `malloc()` in all our projects, the Linux kernel - whose coding style we usually follow - doesn't either. In case you're interested in pro and con arguments for casting:

- [contra](https://stackoverflow.com/questions/1565496/specifically-whats-dangerous-about-casting-the-result-of-malloc/)
- [pro](https://wiki.sei.cmu.edu/confluence/display/c/MEM02-C.+Immediately+cast+the+result+of+a+memory+allocation+function+call+into+a+pointer+to+the+allocated+type)

Reply:
It sounds good and is worth learning.